File: examples/inline/java/grounding/google_search_grounding/003-creating-a-grounded-agent.java
import com.google.adk.agents.LlmAgent;
import com.google.adk.tools.GoogleSearchTool;

LlmAgent rootAgent = LlmAgent.builder()
    .name("google_search_agent")
    .model("gemini-flash-latest")
    .instruction("Answer questions using Google Search when needed. Always cite sources.")
    .description("Professional search assistant with Google Search capabilities")
    .tools(GoogleSearchTool.INSTANCE)
    .build();

File: examples/inline/kotlin/grounding/grounding_with_search/003-creating-a-grounded-agent.kt
import com.google.adk.kt.agents.Instruction
import com.google.adk.kt.agents.LlmAgent
import com.google.adk.kt.models.Gemini
import com.google.adk.kt.tools.VertexAiSearchTool

// Configuration
val DATASTORE_ID =
    "projects/YOUR_PROJECT_ID/locations/global/collections/default_collection/dataStores/YOUR_DATASTORE_ID"

val rootAgent =
    LlmAgent(
        name = "vertex_search_agent",
        model = Gemini(name = "gemini-flash-latest"),
        instruction =
            Instruction(
                "Answer questions using Agent Search to find information from internal " +
                    "documents. Always cite sources when available.",
            ),
        description = "Enterprise document search assistant with Agent Search capabilities",
        tools = listOf(VertexAiSearchTool(dataStoreId = DATASTORE_ID)),
    )